src/extension/providers/dart_reference_provider.ts

```diff
@@ -33,11 +34,16 @@ export class DartReferenceProvider implements ReferenceProvider, DefinitionProvi
 	}
 
 	public async provideDefinition(document: TextDocument, position: Position, token: CancellationToken): Promise<DefinitionLink[] | undefined> {
-		const resp = await this.analyzer.analysisGetNavigation({
-			file: fsPath(document.uri),
-			length: 0,
-			offset: document.offsetAt(position),
-		});
+		let resp1 = this.fileTracker.getNavigationTargets(fsPath(document.uri), document.offsetAt(position));
```

Comment:
nit: Is there any reason to name this `resp1` and not just `resp` and remove the assignment on line 46?

Reply:
Probably not. I just did it to satisfy the type system; if you remove this it seems to want non-null assertions on the usages of `resp` below. That's probably better.